core/src/rpc.rs

```diff
@@ -733,41 +733,58 @@ impl JsonRpcRequestProcessor {
         slot: Slot,
         config: Option<RpcEncodingConfigWrapper<RpcConfirmedBlockConfig>>,
     ) -> Result<Option<UiConfirmedBlock>> {
-        let config = config
-            .map(|config| config.convert_to_current())
-            .unwrap_or_default();
-        let encoding = config.encoding.unwrap_or(UiTransactionEncoding::Json);
-        let transaction_details = config.transaction_details.unwrap_or_default();
-        let show_rewards = config.rewards.unwrap_or(true);
-        let _commitment = config.commitment.unwrap_or_default();
-        if self.config.enable_rpc_transaction_history
-            && slot
+        if self.config.enable_rpc_transaction_history {
+            let config = config
+                .map(|config| config.convert_to_current())
+                .unwrap_or_default();
+            let encoding = config.encoding.unwrap_or(UiTransactionEncoding::Json);
+            let transaction_details = config.transaction_details.unwrap_or_default();
+            let show_rewards = config.rewards.unwrap_or(true);
+            let commitment = config.commitment.unwrap_or_default();
+            check_confirmed_commitment(commitment)?;
+
+            // Block is old enough to be finalized
+            if slot
                 <= self
                     .block_commitment_cache
                     .read()
                     .unwrap()
                     .highest_confirmed_root()
-        {
-            let result = self.blockstore.get_confirmed_block(slot, true);
-            self.check_blockstore_root(&result, slot)?;
-            if result.is_err() {
-                if let Some(bigtable_ledger_storage) = &self.bigtable_ledger_storage {
-                    let bigtable_result = self
-                        .runtime
-                        .block_on(bigtable_ledger_storage.get_confirmed_block(slot));
-                    self.check_bigtable_result(&bigtable_result)?;
-                    return Ok(bigtable_result.ok().map(|confirmed_block| {
+            {
+                let result = self.blockstore.get_confirmed_block(slot, true);
```

Comment:
heh, `confirmed` is overloaded here, this should really be named `get_rooted_block` right?

Reply:
Hehe, yes, exactly. Now that I'm reclaiming `confirmed`, there's no reason to keep this method name. Will rename, thanks!

Reply:
There is a whole rabbit hole to go down here with the confirmed/Confirmed prefix. I contemplated updating the name of the bigtable method as well, but I think I'll save auditing all of that for v1.7-only work.